sklearn/utils/extmath.py

```diff
@@ -833,3 +833,13 @@ def _deterministic_vector_sign_flip(u):
     signs = np.sign(u[range(u.shape[0]), max_abs_rows])
     u *= signs[:, np.newaxis]
     return u
+
+
+def weighted_median(array, sample_weight):
+    sorted_idx = np.argsort(array)
+    sample_weight = np.asarray(sample_weight)
+    weight_cdf = sample_weight[sorted_idx].cumsum()
+    weighted_percentile = (weight_cdf - sample_weight/2.0) / weight_cdf[-1]
```

Comment:
Sorry I too missed this. But still the common tests are failing after changing it to 

```
weighted_percentile = (weight_cdf - sample_weight[sorted_idx]/2.0) / weight_cdf[-1]
```

Just to show the example, the output of weighted _percentile and sorted array is 

```
[ 0.          0.          0.          0.03333333  0.06666667  0.06666667
  0.06666667  0.06666667  0.1         0.16666667  0.23333333  0.3
  0.36666667  0.43333333  0.5         0.56666667  0.63333333  0.7
  0.76666667  0.83333333  0.86666667  0.86666667  0.86666667  0.9
  0.96666667]
[ 0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  1.  1.  1.  1.  1.  1.   1.  1.  1.  1.  1.  1.  1.]
```

and thus giving ``1.0`` as the median even when sample weights are provided. I definitely seem to miss something but not sure.